staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -490,6 +490,10 @@ func (u *Unstructured) GetDeletionGracePeriodSeconds() *int64 {
 }
 
 func (u *Unstructured) SetDeletionGracePeriodSeconds(deletionGracePeriodSeconds *int64) {
+	if deletionGracePeriodSeconds == nil {
+		u.setNestedField(nil, "metadata", "deletionGracePeriodSeconds")
```

Comment:
My PR (https://github.com/kubernetes/kubernetes/pull/51940) removes the value if it's `nil` which I think is the right thing to do in this particular case.

Reply:
Yes, this is the quick fix only for the panic.

Reply:
Ah, we have been round this a few times.  Found it again here: https://github.com/kubernetes/kubernetes/pull/53464/files#r148250781 .  Is really any harder to remove the entry?

Reply:
Just to go on record.  I like removing the key entirely from omitempty fields so serializations are closer to a match.

Reply:
@deads2k the PR #51940 by @ash2k does exactly this. Great cleanup.